### PR TITLE
Depth first search

### PR DESCRIPTION
**Pull request by hitha2000**
PR#2449 
Fixes #

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- depth first search in kotlin
- 

#### Languages Used:
- 
- kotlin

#### Files Added:
- 
- Depth_First_Search.kt

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by somya-kapoor** [comment]
@hitha2000 Please add the issue number in the PR description.

File: Depth_First_Search/Depth_First_Search.kt [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+
+fun main(args: Array<String>) {
+  println("graphs".heading())
+
+  val graph = Graph<String>()
+  graph.apply {
+    // Node / vertex "0"
+    
+    // Node / vertex "
```
Comment (MastersAbh) [inline Depth_First_Search/Depth_First_Search.kt]: Please take dynamic inputs

File: Depth_First_Search/Depth_First_Search.kt [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+     addEdge("0", "1"); 
+    addEdge("0", "2");
```
Comment (MastersAbh) [inline Depth_First_Search/Depth_First_Search.kt]: Fix indentation

File: Depth_First_Search/Depth_First_Search.kt [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  }
+
+  return traversalList.joinToString()
+}
+/* OUTPUT 
+Following is Depth First Traversal
+0 1 2 3
+
+
+*/
```
Comment (MastersAbh) [inline Depth_First_Search/Depth_First_Search.kt]: Please give a blank line at the end of file

File: Depth_First_Search/Depth_First_Search.kt [hunk 4]
```diff
@@ -0,0 +1,101 @@
+
```
Comment (MastersAbh) [inline Depth_First_Search/Depth_First_Search.kt]: Add problem statement at the top of the file

**Comment by MastersAbh** [comment]
@hitha2000 are you working on this ?

**Comment by hitha2000** [comment]
No

**Comment by MastersAbh** [comment]
Sure @hitha2000 Closing this after your confirmation.

**Pull request closed without merging**